src/response.rs

```diff
@@ -465,6 +524,16 @@ impl Response {
         self.body.is_empty()
     }
 
+    /// Returns an optional reference to the `Error` if the response was created from one, or else `None`.
+    pub fn error(&self) -> Option<&Error> {
+        self.error.as_ref().map(|v| v.as_ref())
+    }
+
+    /// Takes the `Error` from the response if one exists, replacing it with `None`.
+    pub fn take_error(&mut self) -> Option<Arc<Error>> {
```

Comment:
Alternatively, this could `Rc::try_unwrap()`, but I'm not really too sure about that.